internal/service/kms/secrets_data_source.go

```diff
@@ -77,13 +87,22 @@ func dataSourceSecretsRead(d *schema.ResourceData, meta interface{}) error {
 				params.EncryptionContext[k] = aws.String(v.(string))
 			}
 		}
+
+		if encryption_algorithm, exists := secret["encryption_algorithm"]; exists {
+			params.EncryptionAlgorithm = aws.String(encryption_algorithm.(string))
+		}
+
 		if grant_tokens, exists := secret["grant_tokens"]; exists {
 			params.GrantTokens = make([]*string, 0)
 			for _, v := range grant_tokens.([]interface{}) {
 				params.GrantTokens = append(params.GrantTokens, aws.String(v.(string)))
 			}
 		}
 
+		if keyid, exists := secret["keyid"]; exists {
+			params.KeyId = aws.String(keyid.(string))
```

Comment:
Since the documentation at [KMS KeyId](https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html#KMS-Encrypt-request-KeyId) uses `KeyId`, I think we can refactor this to use

```suggestion
		if key_id, exists := secret["keyid"]; exists {
			params.KeyId = aws.String(key_id.(string))